salt/cloud/clouds/vmware.py

```diff
@@ -348,7 +352,7 @@ def _edit_existing_network_adapter(network_adapter, new_network_name, adapter_ty
     edited_network_adapter.controllerKey = network_adapter.controllerKey
     edited_network_adapter.unitNumber = network_adapter.unitNumber
     edited_network_adapter.addressType = network_adapter.addressType
-    edited_network_adapter.macAddress = network_adapter.macAddress
+    edited_network_adapter.macAddress = mac_address
```

Comment:
Can you change this to ``edited_network_adapter.macAddress = mac_address if mac_address else network_adapter.macAddress``